custom_components/nefiteasy/climate.py

```diff
@@ -48,11 +48,14 @@ async def async_setup_entry(
     entities = []
 
     client = hass.data[DOMAIN][config_entry.entry_id]["client"]
+    climate = NefitThermostat(client, config_entry.data)
 
-    entities.append(NefitThermostat(client, config_entry.data))
+    entities.append(climate)
 
     async_add_entities(entities, update_before_add=True)
 
+    hass.data[DOMAIN][config_entry.entry_id]["climate"] = climate
```

Comment:
We should not store an entity in hass.data. In general, entities should not depend on each other, the Home Assistant architecture is not intended to be used in that way and it will create all sorts of problems.